controllers/provisioning_controller.go

```diff
@@ -523,8 +534,15 @@ func (r *ProvisioningReconciler) updateProvisioningMacAddresses(ctx context.Cont
 			macs = append(macs, bmh.Spec.BootMACAddress)
 		}
 	}
-	provConfig.Spec.ProvisioningMacAddresses = macs
-	return r.Client.Update(ctx, provConfig)
+
+	if !reflect.DeepEqual(provConfig.Spec.ProvisioningMacAddresses, macs) {
```

Comment:
`macs` should be sorted to avoid unnecessary writes.

Reply:
Ack